glib-macros/src/lib.rs

```diff
@@ -728,8 +728,75 @@ pub fn downgrade(input: TokenStream) -> TokenStream {
 /// assert_eq!(var.get::<Foo>(), Some(v));
 /// ```
 ///
-/// [`glib::Variant`]: variant/struct.Variant.html
-#[proc_macro_derive(Variant)]
+/// Enums are serialized as a tuple `(sv)` with the first value as a [kebab case] string for the
+/// enum variant, or just `s` if this is a C-style enum. Some additional attributes are supported
+/// for enums:
+/// - `#[variant_enum(repr)]` to serialize the enum variant as an integer type instead of `s`.  The
+/// `#[repr]` attribute must also be specified on the enum with a sized integer type, and the type
+/// must implement `Copy`.
+/// - `#[variant_enum(enum)]` uses [`EnumClass`] to serialize/deserialize as nicks. Meant for use
+/// with [`glib::Enum`](Enum).
+/// - `#[variant_enum(flags)]` uses [`FlagsClass`] to serialize/deserialize as nicks. Meant for use
+/// with [`glib::flags`](macro@flags).
+/// - `#[variant_enum(enum, repr)]` serializes as `i32`. Meant for use with [`glib::Enum`](Enum).
+/// The type must also implement `Copy`.
+/// - `#[variant_enum(flags, repr)]` serializes as `u32`. Meant for use with
+/// [`glib::flags`](macro@flags).
+///
+/// # Example
+///
+/// ```
+/// use glib::prelude::*;
+///
+/// #[derive(Debug, PartialEq, Eq, glib::Variant)]
+/// enum Foo {
+///     MyA,
```

Comment:
How is this serialized? Will it be an `s` for this variant and `sv` for the others? That seems annoying to deserialize :)

Reply:
It's `(sv)` for all of the enum variants in `Foo`, that's the `StaticVariantType` and it won't violate that. `s` is only used if the enum only has unit variants, i.e. cases where you could put an integer discriminator after any of the variants.

Reply:
Ok, great :)